### PR TITLE
WIP: RequestManager refactor

### PR DESCRIPTION
**Pull request by ichorid**
This PR completely rewrites TriblerRequestManager into a singleton. This is necessary to support #4750 in sane way. Also, it makes writing requests a bit easier.

### CODE REVIEW

**Comment by ichorid** [comment]
retest this please

File: TriblerGUI/tribler_request_manager.py [hunk 1]
```diff
@@ -215,29 +90,26 @@ def update_worker_settings(self, protocol=None, host=None, port=None, key=None):
...
+                        port=self.default_host,
+                        key=self.default_key,
+                    )
+                )
+        network_reply = self.request_workers[worker_index].do_request(priority, endpoint, reply_callback, data, method)
         request_manager.set_reply_handle(network_reply)
 
     def download_file(self, request_manager, endpoint, reply_callback):
-        self.perform_request(request_manager, endpoint, reply_callback, "", method="GET")
+        self.dispatch_request(request_manager, endpoint, reply_callback, "", method="GET")
```
Comment (qstokkink) [inline TriblerGUI/tribler_request_manager.py]: Shouldn't this take a `priority` argument?

Comment (ichorid) [reply TriblerGUI/tribler_request_manager.py]: Probably. I'm going to refactor this further.

**Comment by ichorid** [comment]
retest this please

**Pull request merged**